src/lib/netlink/bond.rs

```
@@ -92,57 +83,42 @@ const IFLA_BOND_AD_INFO_ACTOR_KEY: u16 = 3;
-                ad_info.actor_key = parse_as_u16(data)
-            }
-            IFLA_BOND_AD_INFO_PARTNER_KEY => {
-                ad_info.partner_key = parse_as_u16(data)
-            }
-            IFLA_BOND_AD_INFO_PARTNER_MAC => {
-                ad_info.partner_mac = parse_as_48_bits_mac(data)
+    for nla in nlas {
+        if let Ok(nla) = nla {
+            match nla.kind(){
+                IFLA_BOND_AD_INFO_AGGREGATOR => {
+                    ad_info.aggregator = parse_as_u16(nla.value())
+                }
+                IFLA_BOND_AD_INFO_NUM_PORTS => {
+                    ad_info.num_ports = parse_as_u16(nla.value())
+                }
+                IFLA_BOND_AD_INFO_ACTOR_KEY => {
+                    ad_info.actor_key = parse_as_u16(nla.value())
+                }
+                IFLA_BOND_AD_INFO_PARTNER_KEY => {
+                    ad_info.partner_key = parse_as_u16(nla.value())
+                }
+                IFLA_BOND_AD_INFO_PARTNER_MAC => {
+                    ad_info.partner_mac = parse_as_48_bits_mac(nla.value())
+                }
+                _ => (),
             }
-            _ => (),
         }
-        i = i + hdr.nla_len;
     }
     ad_info
 }
 
 fn get_bond_mode(raw: &[u8]) -> u8 {
-    let mut i: usize = 0;
-    while i < raw.len() {
-        let hdr_ptr = raw.as_ptr().wrapping_offset(i.try_into().unwrap());
-        let hdr = parse_nla_header(hdr_ptr);
-        let data_ptr = raw
-            .as_ptr()
-            .wrapping_offset((i + NL_ATTR_HDR_LEN).try_into().unwrap());
-        let data = unsafe {
-            slice::from_raw_parts(data_ptr, hdr.nla_len - NL_ATTR_HDR_LEN)
-        };
-        match hdr.nla_type {
-            IFLA_BOND_MODE => return parse_as_u8(data),
-            _ => (),
+    let nlas = NlasIterator::new(raw);
+    for nla in nlas {
+        if let Ok(nla) = nla {
+            if nla.kind() == IFLA_BOND_MODE{
+                return parse_as_u8(nla.value());
+            }
```

Please use eprintln!() for errors instead of silently ignore them.